src/core/lib/resolver/resolver_registry.cc

```diff
@@ -92,17 +96,20 @@ OrphanablePtr<Resolver> ResolverRegistry::CreateResolver(
 
 std::string ResolverRegistry::GetDefaultAuthority(
     absl::string_view target) const {
+  std::string canonical_target;
   URI uri;
-  ResolverFactory* factory = FindResolverFactory(target, &uri);
+  ResolverFactory* factory =
+      FindResolverFactory(target, &uri, &canonical_target);
   if (factory == nullptr) return "";
   return factory->GetDefaultAuthority(uri);
 }
 
 std::string ResolverRegistry::AddDefaultPrefixIfNeeded(
     absl::string_view target) const {
+  std::string canonical_target;
   URI uri;
-  FindResolverFactory(target, &uri);
-  return uri.ToString();
+  FindResolverFactory(target, &uri, &canonical_target);
+  return canonical_target.empty() ? std::string(target) : canonical_target;
```

Comment:
Please add a TODO here that we should fix callers to accept the return value of `uri.ToString()` here and remove the `canonical_target` arg from `FindResolverFactory()`.

Reply:
Done